### PR TITLE
Mfgtool flashlayout

### PR DESCRIPTION
**Pull request by igoropaniuk**

### CODE REVIEW

File: meta-lmp-bsp/recipes-core/images/initramfs-ostree-lmp-recovery.bbappend [hunk 1]
```diff
@@ -31,3 +31,11 @@ fakeroot do_populate_recovery_rootfs_custom () {
 }
 
 IMAGE_PREPROCESS_COMMAND:append:stm32mp1common = " do_populate_recovery_rootfs_custom; "
+
+python __anonymous () {
+    if 'stm32mp1common' in d.getVar('MACHINEOVERRIDES').split(':'):
+        bb.build.addtask('do_create_flashlayout_config', 'do_build', 'do_image_complete', d)
+        # Deploy TF-A binaries before creating flashlayout
+        tfa_pn = d.getVar('PREFERRED_PROVIDER_virtual/trusted-firmware-a')
+        d.appendVarFlag('do_create_flashlayout_config', 'depends', ' %s:do_deploy' % tfa_pn)
```
Comment (quaresmajose) [inline meta-lmp-bsp/recipes-core/images/initramfs-ostree-lmp-recovery.bbappend:40]: I believe `virtual/trusted-firmware-a:do_deploy'` can be used directly here without `tfa_pn`

**Review by quaresmajose — APPROVED** [review]
LGTM with a little suggestion

File: meta-lmp-bsp/recipes-core/images/initramfs-ostree-lmp-recovery.bbappend (hunk 1 above)
Comment (igoropaniuk) [reply meta-lmp-bsp/recipes-core/images/initramfs-ostree-lmp-recovery.bbappend:40]: not really, I see this issue:

```
ERROR: Error for /home/xdev/foundries.io/reps/lmp-test.repo/build-lmp-mfgtool/conf/../../layers/meta-lmp/meta-lmp-base/recipes-core/images/initramfs-ostree-lmp-recovery.bb:do_sdk_depends[depends], dependency virtual/optee-os:do_deployvirtual/trusted-firmware-a:do_deploy in 'initramfs-ostree-lmp-recovery:do_image_cpio initramfs-ostree-lmp-recovery:do_image virtual/optee-os:do_populate_sysroot gdb-cross-arm:do_populate_sysroot openocd-stm32mp-native:do_populate_sysroot stm32wrapper4dbg-native:do_populate_sysroot virtual/kernel:do_deploy virtual/optee-os:do_deployvirtual/trusted-firmware-a:do_deploy opkg-native:do_populate_sysroot opkg-utils-native:do_populate_sysroot makedevs-native:do_populate_sysroot virtual/fakeroot-native:do_populate_sysroot ldconfig-native:do_populate_sysroot virtual/update-alternatives-native:do_populate_sysroot update-rc.d-native:do_populate_sysroot virtual/kernel:do_packagedata pigz-native:do_populate_sysroot meta-extsdk-toolchain:do_populate_sysroot' does not contain exactly one ':' character.
 Task 'depends' should be specified in the form 'packagename:task'

```

Comment (quaresmajose) [reply meta-lmp-bsp/recipes-core/images/initramfs-ostree-lmp-recovery.bbappend:40]: The issue can be the missing space between `virtual/optee-os:do_deployvirtual/trusted-firmware-a:do_deploy`. It should be `d.appendVarFlag('do_create_flashlayout_config', 'depends', ' virtual/trusted-firmware-a:do_deploy')`, is this what you tried?

Comment (igoropaniuk) [reply meta-lmp-bsp/recipes-core/images/initramfs-ostree-lmp-recovery.bbappend:40]: ah yes, I dropped that space, will retest one. The version I tested is:
```
--- a/meta-lmp-bsp/recipes-core/images/initramfs-ostree-lmp-recovery.bbappend
+++ b/meta-lmp-bsp/recipes-core/images/initramfs-ostree-lmp-recovery.bbappend
@@ -36,6 +36,5 @@ python __anonymous () {
     if 'stm32mp1common' in d.getVar('MACHINEOVERRIDES').split(':'):
         bb.build.addtask('do_create_flashlayout_config', 'do_build', 'do_image_complete', d)
         # Deploy TF-A binaries before creating flashlayout
-        tfa_pn = d.getVar('PREFERRED_PROVIDER_virtual/trusted-firmware-a')
-        d.appendVarFlag('do_create_flashlayout_config', 'depends', ' %s:do_deploy' % tfa_pn)
+        d.appendVarFlag('do_create_flashlayout_config', 'depends', 'virtual/trusted-firmware-a:do_deploy')
 }

```

Comment (igoropaniuk) [reply meta-lmp-bsp/recipes-core/images/initramfs-ostree-lmp-recovery.bbappend:40]: fixed

**Review by ricardosalveti — APPROVED** [review]
LGTM

**Pull request merged**